vowpalwabbit/gd.cc

```diff
@@ -731,82 +735,95 @@ void save_load_online_state(vw& all, io_buf& model_file, bool read, bool text, g
   }
 
   uint64_t length = (uint64_t)1 << all.num_bits;
-  uint64_t stride = (uint64_t)1 << all.reg.stride_shift;
 
-  int c = 0;
-  uint64_t i = 0;
+  weight_parameters& weights = all.weights;
+  uint32_t i = 0;
   size_t brw = 1;
   do
   { brw = 1;
-    weight* v;
+    weight_parameters::iterator v = weights.begin(0);
+    weight_parameters::iterator v1 = weights.begin(1);
```

Comment:
Or better: (&(*v))[1]  
(it looks like formatting was messed up in the first suggestion)